pyro/contrib/gp/parameterized.py

```diff
@@ -162,7 +170,7 @@ def autoguide(self, name, dist_constructor):
         if dist_constructor is dist.Delta:
             p_map = Parameter(p.detach())
             self.register_parameter("{}_map".format(name), p_map)
-            self.set_constraint("{}_map".format(name), self._priors[name].support)
+            self.set_constraint("{}_map".format(name), _get_independent_support(self._priors[name]))
```

Comment:
we don't need to create `unconstrained` parameter for real independent constraint.